libs/blocks/global-navigation/global-navigation.js

```diff
@@ -488,8 +477,11 @@ class Gnav {
       const decorateDropdown = async () => {
         template.removeEventListener('click', decorateDropdown);
         clearTimeout(decorationTimeout);
-        await this.loadDecorateDropdown();
-        this.decorateDropdown({
+
+        const menuLogic = await loadDecorateMenu();
+        this.decorateMenu = menuLogic.decorateMenu;
+
+        this.decorateMenu({
```

Comment:
curious why the assignment to this.decorateMenu as it doesn't appear to be used elsewhere. Could you just use:
```suggestion
        menuLogic.decorateMenu({
```